qiita_db/archive.py

```diff
@@ -153,9 +172,9 @@ def retrieve_feature_values(cls, archive_merging_scheme=None,
             else:
                 qdb.sql_connection.TRN.add(sql.format(''))
 
-            return {k: v for k, v in
-                    qdb.sql_connection.TRN.execute_fetchindex()}
+            return dict(qdb.sql_connection.TRN.execute_fetchindex())
 
+    @classmethod
     def insert_features(self, merging_scheme, features):
```

Comment:
I hate to sound picky, but if insert_features is going to be a class method, can you change all instances of self to cls, just for convention's sake?

Reply:
done.

Reply:
thanks! Will merge as soon as testing is completed.